Automattic/wp-calypso
# Staging sites: fix staging creation navigation bug

- Related to https://github.com/Automattic/dotcom-forge/issues/2208

## Proposed Changes

When the user creates a new staging site, and move back and forth to the page while the staging creation is in progress, faces strange issues like:

- The button to create a site becomes visible again
- When the simple staging site has been created, an error message is displayed instead of the progress bar.

This pr fixes the above issues.

NOTE: This doesn't fix the case where the user reloads the page. Instead for that case, we have implemented a lock mechanism in D109053 that prevents the user to create a new site. It definitely not a perfect solution, but it's better UX-wise solution than before.


## Testing Instructions

1. Create a new site with a business plan and make it atomic.
2. Use this diff D109294-code
3. Navigate to the `/navigation` page.
4. Press the "Add Staging Site" button.
5. Imediatelly, press another link in the sidebar on the left, eg: "Pages"
6. Navigate back to `/hosting` using the sidebar.
7. Ensure that you see the progress bar.

| **Before** | **After** |
| :---:         |    :---:       |
|![before1](https://user-images.githubusercontent.com/497103/234611786-15abfaf3-94d4-4e53-8d68-9bcc9b0bc772.png)![before2](https://user-images.githubusercontent.com/497103/234611802-9c41b3c9-b33a-4382-b523-f745dc2c2c3c.png)| ![after1](https://user-images.githubusercontent.com/497103/234611694-3ca789c4-bf87-43df-9fd7-4a413251760a.png)|




## Pre-merge Checklist
